frame/evm/precompile/blake2/src/lib.rs

```diff
@@ -46,16 +45,16 @@ impl LinearCostPrecompile for Blake2F {
 
 		let mut rounds_buf: [u8; 4] = [0; 4];
 		rounds_buf.copy_from_slice(&input[0..4]);
-		let rounds: u32 = u32::from_le_bytes(rounds_buf);
+		let rounds: u32 = u32::from_be_bytes(rounds_buf);
 
```

Comment:
FYI, these changes (the array indexing and endianness ones) are probably useful outside of this PR.

Reply:
Please review the endianness changes thoroughly :)

Reply:
Looks right. Our original code is wrong.